adal/src/main/java/com/microsoft/aad/adal/ExceptionExtensions.java

```diff
@@ -34,19 +33,19 @@
     private ExceptionExtensions() {
         // Intentionally left blank
     }
-    static String getExceptionMessage(Exception ex) {
+
+    static String getExceptionMessage(Throwable ex) {
         String message = null;
 
         if (ex != null) {
             message = ex.getMessage();
 
             if (message == null) {
-                final StringWriter sw = new StringWriter();
-                ex.printStackTrace(new PrintWriter(sw));
-                message = sw.toString();
+                message = Log.getStackTraceString(ex);
```

Comment:
Are we going to log the entire stack trace? I'm concerned on the size of the telemetry.

Reply:
This is sort of an evolving discussion. From testing, the size of the telemetry payload is ~67% larger if the broker fails to bind. Per discussion w/ @addev-ashish and @nazukj this isn't something I'm going to spend time optimizing right now